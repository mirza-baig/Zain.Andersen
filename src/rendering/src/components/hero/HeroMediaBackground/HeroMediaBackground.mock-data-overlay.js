const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'HeroMediaBackground',
  },
  fields: {
    overlay: {
      id: 'cec0af91-e32d-4183-86b2-1594757d8052',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Overlay/Black',
      name: 'Black',
      displayName: 'Black',
      fields: {
        Value: {
          value: 'black',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    theme: {
      value: '',
    },
    eyebrowLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eyebrowText: {
      value: '',
    },
    headlineText: {
      value: 'Join our team',
    },
    body: {
      value: "<p>Ready to make a difference? We're excited to meet you.</p>",
    },
    fontColor: null,
    cta1Icon: {
      id: '0db14183-ef30-4ccd-9781-a27d5a7c893b',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/External-Link',
      name: 'External Link',
      displayName: 'External Link',
      fields: {
        Value: {
          value: 'external-link',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Link: {
      value: {
        href: 'http://andersencareers.com/',
        text: 'Explore Career Opportunities',
        linktype: 'external',
        url: 'http://andersencareers.com/',
        anchor: '',
        target: '_blank',
      },
    },
    cta1Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'primary',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    primaryImage: {
      value: {
        src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/why-andersen-hero-image1400x617.jpg?h=617&iar=0&w=1440',
        alt: 'why-andersen-hero-image1400x617',
        width: '1440',
        height: '617',
      },
    },
    primaryImageMobile: {
      value: {},
    },
    primaryImageMobileFocusArea: null,
    primaryVideo: {
      id: '01ff5d6d-b590-4f4f-b6d9-770d006de994',
      url: '/Data/Elements/Videos/YouTube-Video',
      name: 'YouTube Video',
      displayName: 'YouTube Video',
      fields: {
        youTubeAutoLoop: {
          value: false,
        },
        youTubeAutoPlay: {
          value: true,
        },
        youTubeClosedCaptions: {
          value: true,
        },
        youTubeDisableKeyboard: {
          value: false,
        },
        youTubeModestBranding: {
          value: false,
        },
        youTubeMute: {
          value: false,
        },
        youTubeShowControls: {
          value: true,
        },
        videoLazyLoad: {
          value: false,
        },
        videoWidth: {
          value: 600,
        },
        videoHeight: {
          value: 338,
        },
        videoId: {
          value: 'K-eDe8GDMUo',
        },
      },
      templateId: '10ffc85c-56a6-4a99-953c-56bb3405c1e6',
      templateName: 'YouTube Video',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
