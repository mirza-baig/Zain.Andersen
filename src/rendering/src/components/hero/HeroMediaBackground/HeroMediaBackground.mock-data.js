const defaultData = {
    rendering: {
        dataSource: 'sampledatasource',
        componentName: 'HeroMediaBackground',
    },
    
  fields: {
    overlay: {
      id: "5086f411-a48f-410f-8b24-580d39057406",
      url: "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Overlay/None",
      name: "None",
      displayName: "None",
      fields: {
        Value: {
          value: ""
        }
      },
      templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
      templateName: "Enum"
    },
    componentSpacing: null,
    sectionId: {
      value: ""
    },
    theme: {
      value: ""
    },
    eyebrowText: {
      value: "About us"
    },
    eyebrowLevel: {
      id: "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
      url: "http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
      name: "Heading 2",
      displayName: "Heading 2",
      fields: {
        Value: {
          value: "h2"
        }
      },
      templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
      templateName: "Enum"
    },
    headlineText: {
      value: "AMERICA'S MOST LOVED BRAND OF WINDOWS AND DOORS"
    },
    body: {
      value: "<p>2022 Andersen Brand Surveys of U.S. realtors, contractors, builders &amp; homeowners</p>"
    },
    fontColor: {
      id: "12a26c7b-7f96-4c40-9762-187d4132dd6b",
      url: "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Font-Color/White",
      name: "White",
      displayName: "White",
      fields: {
        Value: {
          value: "white"
        }
      },
      templateId: "d2923fee-da4e-49be-830c-e27764dfa269",
      templateName: "Enum"
    },
    cta1Icon: null,
    cta1Link: {
      value: {
        href: ""
      }
    },
    cta1Style: null,
    primaryImageMobile: {
      value: {}
    },
    primaryImageMobileFocusArea: null,
    primaryImage: {
      value: {
        src: "https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/about-hero-image-1440x617-(1).jpg?h=617&iar=0&w=1440",
        alt: "about-hero-image-1440x617",
        width: "1440",
        height: "617"
      }
    },
    primaryVideo: null,
    eventName: {
      value: ""
    },
    eventType: {
      value: ""
    },
    eventZone: {
      value: ""
    }
  }
};

export default defaultData;
