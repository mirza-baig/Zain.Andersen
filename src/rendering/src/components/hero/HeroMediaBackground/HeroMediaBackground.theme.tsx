// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const HeroMediaBackgroundTheme: ThemeFile = {
  aw: {
    classes: {
      /** Theme classes **/
      mediaContainer: 'relative max-md:mb-s ',
      overlay: 'absolute z-[0] w-full h-full bg-[rgba(0,0,0,0.5)] left-0 top-0 bottom-20',
      iconWrapper: 'mr-s mb-s md:mr-l md:mb-l',
      contentWrapper: 'max-md:px-m md:absolute w-full h-full z-[1] flex items-center md:top-0 ',
      contentContainer: 'max-w-screen-xl md:ml-[8%] 2xl:ml-auto flex flex-col basis-full mx-auto ',
      eyebrow: 'my-xxxs md:max-w-[585px] texy-sm-xs md:text-xs font-medium uppercase font-sans',
      headline: 'my-xxxs md:max-w-[585px] text-sm-l md:text-l font-heavy',
      bodyClass: 'my-xxxs md:max-w-[585px] text-body',
      buttonGroupClass: {
        wrapper: 'my-s md:my-xs',
        cta1Classes: 'text-black',
      },
    },
  },
  rba: {
    classes: {
      /** Theme classes **/
      mediaContainer: 'relative max-md:mb-s',
      overlay: 'absolute z-[1] w-full h-full bg-[rgba(0,0,0,0.5)] left-0 top-0',
      iconWrapper: 'mr-s mb-s md:mr-ml md:mb-ml',
      contentWrapper: 'max-md:px-m md:absolute w-full h-full z-[1] flex items-center md:top-0',
      contentContainer: 'max-w-screen-xl md:ml-[8%] 2xl:ml-auto flex flex-col basis-full mx-auto',
      eyebrow: 'my-xxxs text-link md:max-w-[490px] font-semi-bold uppercase',
      headline: 'my-xxxs text-sm-s md:text-xxl md:max-w-[490px] font-medium',
      bodyClass: 'my-xxxs md:max-w-[490px] text-large-body',
      buttonGroupClass: {
        wrapper: 'my-s md:my-xs',
        cta1Classes: 'md:bg-white md:text-black',
      },
    },
  },
};
