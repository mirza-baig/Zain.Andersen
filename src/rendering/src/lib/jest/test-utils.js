// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
