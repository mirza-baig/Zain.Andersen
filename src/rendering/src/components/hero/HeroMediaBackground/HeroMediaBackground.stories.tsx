// Global
import { Story, Meta } from '@storybook/react';
// Local
import HeroMediaBackground, { HeroMediaBackgroundProps } from './HeroMediaBackground';
import defaultData from './HeroMediaBackground.mock-data';
import overlayData from './HeroMediaBackground.mock-data-overlay';

export default {
  title: 'Hero/HeroMediaBackground',
  component: HeroMediaBackground,
} as Meta;
const Template: Story<HeroMediaBackgroundProps> = (props) => <HeroMediaBackground {...props} />;

export const Default = Template.bind({});
Default.args = defaultData;
export const Overlay = Template.bind({});
Overlay.args = overlayData;
