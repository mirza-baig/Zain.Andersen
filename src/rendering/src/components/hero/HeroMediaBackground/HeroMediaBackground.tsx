import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import classNames from 'classnames';
import { useTheme } from 'lib/context/ThemeContext';
import { useState } from 'react';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';

// Components
import { ComponentProps } from 'lib/types/component-props';
import { getEnum } from 'lib/utils';
import { useCurrentScreenType } from 'lib/utils/get-screen-type';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { GetComponentSpacingClass } from 'src/helpers/Component/Component';
import { Eyebrow } from 'src/helpers/Eyebrow';
import { Headline } from 'src/helpers/Headline';
import { MediaPrimary } from 'src/helpers/Media';
import RichTextWrapper from 'src/helpers/RichTextWrapper/RichTextWrapper';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { HeroMediaBackgroundTheme } from './HeroMediaBackground.theme';
import { cta1ToButtonProps, cta2ToButtonProps } from 'src/helpers/Button';

export type HeroMediaBackgroundProps =
  Feature.EnterpriseWeb.Enterprise.Components.Hero.HeroMediaBackground & ComponentProps;
const HeroMediaBackground = (props: HeroMediaBackgroundProps) => {
  const { fields } = props;

  const { themeName, themeData } = useTheme(HeroMediaBackgroundTheme);
  const { screenType } = useCurrentScreenType();

  const [isVideoPaused, setIsVideoPaused] = useState(
    !fields?.primaryVideo?.fields.youTubeAutoPlay?.value || false
  );

  if (!fields) {
    return null;
  }

  const isDesktop = screenType !== 'sm' ? true : false;
  let hasOverlay = getEnum(fields.overlay) ? true : false;

  let fontColorClass = '';

  if (hasOverlay) {
    fontColorClass = 'md:text-white';
  } else {
    switch (getEnum(fields.fontColor)) {
      case 'gray':
        fontColorClass = 'md:text-dark-gray';
        break;
      case 'white':
        fontColorClass = 'md:text-white';
        break;
      default:
        fontColorClass = 'md:text-black';
        break;
    }
  }

  if (themeName === 'rba') {
    if (fields.overlay === null) {
      hasOverlay = true;
    }
  }

  const togglePlayPause = () => {
    const videoId = (
      props.fields?.primaryVideo as Feature.EnterpriseWeb.Enterprise.Elements.Media.YouTubeVideo
    )?.fields.videoId.value;

    const iframeContentWindow = (document.getElementById(videoId) as HTMLIFrameElement)
      ?.contentWindow;

    if (iframeContentWindow) {
      iframeContentWindow.postMessage(
        `{"event":"command", "func":"${isVideoPaused ? 'playVideo' : 'pauseVideo'}", "args":""}`,
        '*'
      );

      setIsVideoPaused(!isVideoPaused);
    }
  };
  return (
    <div
      data-component="hero/heromediabackground"
      className={classNames(
        'relative',
        GetComponentSpacingClass(props?.fields?.componentSpacing, 'none')
      )}
    >
      <div className={themeData.classes.mediaContainer}>
        <MediaPrimary
          imageLayout="fill"
          ratio="video"
          additionalDesktopClasses="h-[280px] md:h-[620px]"
          additionalMobileClasses="h-[280px]"
          priority
          {...props}
        />
        {fields.primaryVideo && (
          <div
            className={classNames(
              'absolute right-0 bottom-0 z-10 cursor-pointer rounded-full bg-black',
              themeData.classes.iconWrapper
            )}
            onClick={togglePlayPause}
          >
            {isVideoPaused ? <SvgIcon icon="play" size="lg" /> : <SvgIcon icon="pause" size="lg" />}
          </div>
        )}
      </div>
      {hasOverlay && isDesktop && <div className={themeData.classes.overlay}></div>}
      <div className={classNames(themeData.classes.contentWrapper, fontColorClass)}>
        <div className={classNames(themeData.classes.contentContainer, fontColorClass)}>
          <Eyebrow classes={classNames(themeData.classes.eyebrow, fontColorClass)} {...props} />
          <Headline
            classes={classNames(themeData.classes.headline, fontColorClass)}
            {...props}
            useTag="h1"
          />
          <RichTextWrapper
            classes={classNames(themeData.classes.bodyClass, fontColorClass)}
            field={props.fields?.body}
          />
          {fields.cta1Link && fields.cta1Link.value.href && (
            <ButtonGroup
              cta1={cta1ToButtonProps(props, themeData.classes.buttonGroupClass.cta1Classes)}
              cta2={cta2ToButtonProps(props, themeData.classes.buttonGroupClass.cta2Classes)}
              wrapperClasses={themeData.classes.buttonGroupClass.wrapper}
            />
          )}
        </div>
      </div>
    </div>
  );
};

export default withDatasourceCheck()<HeroMediaBackgroundProps>(HeroMediaBackground);
