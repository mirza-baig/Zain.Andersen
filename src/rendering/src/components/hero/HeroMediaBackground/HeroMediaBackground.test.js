// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import HeroMediaBackground from './HeroMediaBackground';
import defaultData from './HeroMediaBackground.mock-data';

it('renders correctly', async () => {
  const component = snapshot(HeroMediaBackground, { componentProps: defaultData });
  await hasDataComponent(component, 'hero/heromediabackground');
});
