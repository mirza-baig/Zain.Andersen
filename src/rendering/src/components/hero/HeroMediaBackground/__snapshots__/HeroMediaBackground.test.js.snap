// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="relative my-0"
  data-component="hero/heromediabackground"
>
  <div
    class="relative max-md:mb-s"
  >
    <div
      class="relative mx-auto hidden md:block aspect-hero   h-[280px] md:h-[620px]"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="about-hero-image-1440x617"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=3840&q=75"
          srcset="/_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </span>
    </div>
    <div
      class="relative w-full md:hidden h-[280px]"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="about-hero-image-1440x617"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=3840&q=75"
          srcset="/_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fedge.sitecorecloud.io%2Fandersencorporation-m9n8s20s%2Fmedia%2FProject%2FAndersenCorporation%2FAndersenWindows%2FAndersenWindows%2Fabout-hero-image-1440x617-(1).jpg%3Fh%3D617%26iar%3D0%26w%3D1440&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
        />
      </span>
    </div>
  </div>
  <div
    class="max-md:px-m md:absolute w-full h-full z-[1] flex items-center md:top-0 md:text-white"
  >
    <div
      class="max-w-screen-xl md:ml-[8%] 2xl:ml-auto flex flex-col basis-full mx-auto md:text-white"
    >
      <div
        class="my-xxxs md:max-w-[585px] texy-sm-xs md:text-xs font-medium uppercase font-sans md:text-white"
      >
        <h2>
          About us
        </h2>
      </div>
      <div
        class="items-top font-sans [&_a:hover]:underline my-xxxs md:max-w-[585px] text-sm-l md:text-l font-heavy md:text-white"
      >
        <h1>
          AMERICA'S MOST LOVED BRAND OF WINDOWS AND DOORS
        </h1>
      </div>
      <div
        class="body-copy my-xxxs md:max-w-[585px] text-body md:text-white"
      >
        <div>
          <p>
            2022 Andersen Brand Surveys of U.S. realtors, contractors, builders & homeowners
          </p>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
